### DISCUSSION TITLE
Web Adaptors Fail to configure

### DISCUSSION THREAD

**Discussion by wmedernach** [Esri/arcgis-powershell-dsc#512] [Q&A]
I've just installed a Single Machine Base Deployment.  I had originally modified the BaseDeployment-SingleMachine.json from the 4.1.0 folder, but this was running with 4.2.0 so I had to add the WebAdaptorConfig section at the top.

Everything seemed to install fine, but it appears that the WAs were not configured.  The admin URLs with the ports work, but when you try to goto the WA it displays the WA configure pages.

Any suggestions?

Log entries showing WAs installed successfully:
12/21/2023 9:25:18 PM: [<machinename>]:[[ArcGIS_Install]WebAdaptorIIS-Portal-portalInstall] WebAdaptorIIS-Portal-portal installation was successful!
12/21/2023 9:26:15 PM: [<machinename>]:[[ArcGIS_Install]WebAdaptorIIS-Server-arcgisInstall] WebAdaptorIIS-Server-arcgis installation was successful!

Log entries for WA config:
12/21/2023 9:47:37 PM: [<machinename>]:[[ArcGIS_WebAdaptor]ConfigurePortalWebAdaptor<machinename>-portal] https://<machinename>/portal/webadaptor
12/21/2023 9:47:42 PM: [<machinename>]:[[ArcGIS_WebAdaptor]ConfigurePortalWebAdaptor<machinename>-portal] Output:- Portal configuration requires fully qualified domain name in the WebAdaptor URL.
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Set      ]  [[ArcGIS_WebAdaptor]ConfigurePortalWebAdaptor<machinename>-portal]  in 2.5250 seconds.
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Resource ]  [[ArcGIS_WebAdaptor]ConfigurePortalWebAdaptor<machinename>-portal]
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ Start  Resource ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis]
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ Start  Test     ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis]
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Test     ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis]  in 0.0100 seconds.
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ Start  Set      ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis]
12/21/2023 9:47:42 PM: [<machinename>]:[[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis] https://<machinename>/arcgis/webadaptor
12/21/2023 9:47:42 PM: [<machinename>]:[[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis] Output:- ArcGIS Server configuration requires fully qualified domain name in the WebAdaptor URL.
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Set      ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis]  in 0.1600 seconds.
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Resource ]  [[ArcGIS_WebAdaptor]ConfigureServerWebAdaptor<machinename>-arcgis]
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Set      ]
12/21/2023 9:47:42 PM: [<machinename>]: LCM:  [ End    Set      ]    in  5.8220 seconds.
12/21/2023 9:47:42 PM: Operation 'Invoke CimMethod' complete.

**Comment by cameronkroeker** [comment]
Hi @wmedernach,

In the json configuration file is there an SSL certificate specified for the Web Adaptor? If so then the module will use the value specified for `SslCertificates.CNameFQDN` attribute, otherwise it will use the value returned by `Get-FQDN`. If specifying an SSL certificate be sure to use an FQDN for the `SslCertificates.CNameFQDN` attribute. If not, then on the Target Node can you run the following command?

`Get-FQDN $Node.NodeName`

> Note: Replace $Node.NodeName with the value that is defined for NodeName within the json configuration file.

This could be returning a short name rather than an FQDN.

Thanks,
Cameron K.
